packages/camera/camera/example/ios/RunnerTests/SampleBufferQueueTests.m

```diff
@@ -0,0 +1,37 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera;
+@import camera.Test;
+@import AVFoundation;
+@import XCTest;
+#import <OCMock/OCMock.h>
+
+@interface SampleBufferQueueTests : XCTestCase
+
+@end
+
+@implementation SampleBufferQueueTests
+
+- (void)testSampleBufferCallbackQueueMustBeCaptureSessionQueue {
+  id inputMock = OCMClassMock([AVCaptureDeviceInput class]);
+  OCMStub([inputMock deviceInputWithDevice:[OCMArg any] error:[OCMArg setTo:nil]])
+      .andReturn(inputMock);
+
+  id sessionMock = OCMClassMock([AVCaptureSession class]);
+  OCMStub([sessionMock alloc]).andReturn(sessionMock);
```

Comment:
If you stub a class method you need to call `-stopMock` explicitly or the mock will leak between tests, it won't stop mocking when the mock object deallocs.

Reply:
gotcha. good to know!

Reply:
Ah looks like `stopMocking` is called on dealloc, from the doc: https://ocmock.org/reference/

> The class can be returned to its original state by calling stopMocking. This is only necessary if the original state must be restored before the end of the test. The mock automatically calls stopMocking during its own deallocation.

Reply:
Wrote a dummy test and verified it.

Reply:
I didn't think it works stubbing _class_ methods, `alloc` in this case.  Is that what you tested?

Reply:
Yep, i tested `alloc`. 
![Screen Shot 2022-01-28 at 2 28 23 PM](https://user-images.githubusercontent.com/41930132/151630195-a81937f6-f316-4510-8aea-adbd0dba2f11.png)